astropy/constants/codata2017.py

```diff
@@ -0,0 +1,117 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Astronomical and physics constants in SI units.  See :mod:`astropy.constants`
+for a complete listing of constants defined in Astropy.
+"""
+import math
+
+from .constant import Constant, EMConstant
+
+
+# PHYSICAL CONSTANTS
+
+class CODATA2017(Constant):
+    default_reference = 'CODATA 2017'
+    _registry = {}
+    _has_incompatible_units = set()
+
+
+class EMCODATA2017(CODATA2017, EMConstant):
+    _registry = CODATA2017._registry
+
+
+# Table 3 from Newell, D. B., et al. 2018, Metrologia, 55, L13
+# http://iopscience.iop.org/article/10.1088/1681-7575/aa950a/meta
+
+h = CODATA2017('h', "Planck constant", 6.62607015e-34,
+               'J s', 0.0, system='si')
```

Comment:
I also did not address apparent unit change. For example, `h` is redefined to be in the units of `kg m2 s−1` in order for `kg` to be redefined. Here, it remains `J s` until we figure out how to handle the SI change over at `astropy.units`. Or maybe we don't have to for our purpose (astronomy)? Not sure.

Reply:
I think for the purpose of the constant, it should remain `J s` - we can worry about `kg` later.